planner/core/task.go

```diff
@@ -1347,6 +1347,23 @@ func (p *PhysicalTopN) addPartialLimitForSubScans(copSubPlans []PhysicalPlan, fi
 	return limitAddedPlan
 }
 
+func (p *PhysicalExpand) attach2Task(tasks ...task) task {
+	t := tasks[0].copy()
+	// current expand can only be run in MPP TiFlash mode.
+	if mpp, ok := t.(*mppTask); ok {
+		p.SetChildren(mpp.p)
+		mpp.p = p
+		return mpp
+	}
+	// future code for TiDB self Expand implementation.
```

Comment:
Why not just give error for rootTask? Future code seems useless for now.

Reply:
make sense